src/Appwrite/Platform/Workers/Builds.php

```diff
@@ -207,7 +207,66 @@ protected function buildDeployment(Device $deviceForFunctions, Func $queueForFun
         }
 
         try {
-            if ($isNewBuild && $isVcsEnabled) {
+            if($isNewBuild && !$isVcsEnabled) {
+                // Non-vcs+Template
+
+                $templateRepositoryName = $template->getAttribute('repositoryName', '');
+                $templateOwnerName = $template->getAttribute('ownerName', '');
+                $tempalteVersion = $template->getAttribute('version', '');
+
+                $templateRootDirectory = $template->getAttribute('rootDirectory', '');
+                $templateRootDirectory = \rtrim($templateRootDirectory, '/');
+                $templateRootDirectory = \ltrim($templateRootDirectory, '.');
+                $templateRootDirectory = \ltrim($templateRootDirectory, '/');
+
+                if (!empty($templateRepositoryName) && !empty($templateOwnerName) && !empty($tempalteVersion)) {
```

Comment:
I Dont see an else for this... We should throw an exception if this is the case

Reply:
Not every new build has to have template.